src/muc/mod_muc_room.erl

```diff
@@ -900,7 +900,12 @@ can_send_broadcasts(Role, StateData) ->
 
 broadcast_room_packet(From, FromNick, Role, Packet, StateData) ->
     Activity = get_user_activity(From, StateData),
-    TS = Activity#activity.message_time,
+    TS = case Activity#activity.message_time of
+            0 ->
+                erlang:system_time(microsecond);
+            Value ->
+                Value
```

Comment:
...and is this really not covered by tests? I don't understand, because in normal operation this shouldn't be zero, right?

Reply:
We could set the timestamp here:

```erlang
get_user_activity(JID, StateData) ->
    ?LOG_ERROR(#{what => get_user_activity, jid => JID, activity => StateData#state.activity}),
    case treap:lookup(jid:to_lower(JID), StateData#state.activity) of
    {ok, _P, A} -> A;
    error ->
        MessageShaper = mongoose_shaper:new(get_opt(StateData, user_message_shaper)),
        PresenceShaper = mongoose_shaper:new(get_opt(StateData, user_presence_shaper)),
        #activity{message_shaper = MessageShaper,
                  presence_shaper = PresenceShaper,
                   message_time = erlang:system_time(microsecond)} %% HERE
    end
```

Reply:
Thanks for the clarification, I misunderstood how this was supposed to work.